examples/ota-requestor-app/ota-requestor-common/BDXDownloader.h

```diff
@@ -22,14 +22,32 @@
 
 #pragma once
 
+// Callback to handle received data
+typedef void(*OnBdxBlockReceived)(void * context, chip::bdx::TransferSession::BlockData & blockdata);
+typedef void(*OnBdxTransferComplete)(void * context);
+typedef void(*OnBdxTransferFailed)(void *context);
+
+struct BdxDownloaderCallbacks
```

Comment:
I'm a little worried about this callback interface - specifically that we might end up needing a callback for every BDX message or every BDX event. But I think this is ok for now. I can't think of a better way yet